components/collector/src/source_collectors/dependency_track/source_up_to_dateness.py

```diff
@@ -87,3 +91,21 @@ def _timestamp_to_datetime(timestamp: int | None) -> datetime | None:
     def _is_latest(project: DependencyTrackProject) -> bool:
         """Return whether the project is the latest version."""
         return project.get("isLatest", False)
+
+    def _up_to_dateness(self, project: DependencyTrackProject) -> str:
+        """Return the up-to-dateness of the project."""
+        date_times = self._project_event_datetimes(project)
+        if not date_times:
+            return "unknown"
+        min_date_time = self.days(self.minimum(date_times))
+        if self._target_met(min_date_time, "target"):
+            return "yes"
+        if self._target_met(min_date_time, "near_target"):
+            return "almost"
+        return "no"
+
+    def _target_met(self, value: int, target_type: Literal["target", "near_target"]) -> bool:
+        """Return whether the value meets the target value."""
```

Comment:
Since this method is only used by up to dateness, the direction should always be `<` right? Now it looks like this is actually quite a generic method that belongs in a collector baseclass

Reply:
You're right, but unfortunately the user can change the direction of the source up-to-dateness metric. We have no way (yet) to make changing the direction impossible for certain metric types.

Reply:
I'll add a comment.